bwmenu

```diff
@@ -94,12 +100,30 @@ rofi_menu() {
 show_items() {
   if item=$(
     echo "$ITEMS" \
-    | jq -r ".[] | select( has( \"login\" ) ) | .name" \
+    | jq -r ".[] | select( has( \"login\" ) ) | \"\\(.name)\"" \
+    | dedup_lines \
+    | rofi_menu
+  ); then
+    item_array="$(array_from_name "$item")"
+    copy_password "$item_array"
+  else
+    on_rofi_exit $? "$item"
+  fi
+}
+
+# Similar to show_items() but with more useful information
+show_full_items() {
+  if item=$(
+    echo "$ITEMS" \
+      | jq -r ".[] | select( has( \"login\" )) | \"\\(.id): name: \\(.name), username: \\(.login.username)\"" \
     | rofi_menu
   ); then
-    item=$(echo "$ITEMS" | jq -r ".[] | select(.name == \"$item\")")
+    item_id=$(echo $item | cut -d':' -f1)
+    item=$(echo "$ITEMS" | jq -r ". | map(select(.id == \"$item_id\"))")
     copy_password "$item"
   else
+    item_id=$(echo $item | cut -d':' -f1)
+    item=$(echo "$ITEMS" | jq -r ". | map(select(.id == \"$item_id\"))")
     on_rofi_exit $? "$item"
```

Comment:
Here is a problem, the `on_rofi_exit()` function expects parameter 2 to be the *name* of the item (not the full item as in this case), which is not the unique identifier and so we would end up in an indefinite loop.. We could however change the on_rofi_exit to accept the *id* and edit the *non-duplicate* functions to pass on the id as well and not the name.
What are your thoughts?

Reply:
Right, good catch. Also, `$?` is the value of the prevous command's exit code, not rofi's. I'll put that in a separate variable so no one gets confused.
Opinion-wise, I don't think `copy_password()` and `auto_type()` should go look into the list of all items. My idea was to pass the array of results `item_array` to `on_rofi_exit()` but somehow didn't. In this scenario, I would call `array_from_name()`, and pass the result to `on_rofi_exit()` in `show_items()`. Do you have any preference doing this instead of passing only the id?